lib/usdUfe/ufe/UsdUndoDuplicateCommand.cpp

```diff
@@ -0,0 +1,159 @@
+//
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "UsdUndoDuplicateCommand.h"
+
+#include <usdUfe/base/tokens.h>
+#include <usdUfe/ufe/Global.h>
+#include <usdUfe/ufe/Utils.h>
+#include <usdUfe/undo/UsdUndoBlock.h>
+#include <usdUfe/utils/editRouter.h>
+#include <usdUfe/utils/editRouterContext.h>
+#include <usdUfe/utils/layers.h>
+#include <usdUfe/utils/loadRules.h>
+#include <usdUfe/utils/mergePrims.h>
+
+#include <pxr/base/tf/token.h>
+#include <pxr/usd/sdf/copyUtils.h>
+#include <pxr/usd/usd/prim.h>
+#include <pxr/usd/usd/stage.h>
+
+namespace USDUFE_NS_DEF {
+
+// Ensure that UsdUndoDuplicateCommand is properly setup.
+static_assert(std::is_base_of<Ufe::UndoableCommand, UsdUndoDuplicateCommand>::value);
+static_assert(std::has_virtual_destructor<UsdUndoDuplicateCommand>::value);
+static_assert(!std::is_copy_constructible<UsdUndoDuplicateCommand>::value);
+static_assert(!std::is_copy_assignable<UsdUndoDuplicateCommand>::value);
+static_assert(!std::is_move_constructible<UsdUndoDuplicateCommand>::value);
+static_assert(!std::is_move_assignable<UsdUndoDuplicateCommand>::value);
+
+namespace {
+
+// Local ufePathToPrim that uses input stage instead of getStage function.
+UsdPrim ufePathToPrim(const Ufe::Path& path, const PXR_NS::UsdStageWeakPtr& stage)
```

Comment:
Again, I think a function that is going to use the stage should use a RefPtr, not  WeakPtr. Otehrwise, technically, the check for non-null at entry is useless, although practically it will work because it is unlikely to become invalid during the function...